cgroups/fs/apply_raw.go

```diff
@@ -88,6 +91,93 @@ func Apply(c *cgroups.Cgroup, pid int) (map[string]string, error) {
 	return paths, nil
 }
 
+// Stop stops all processes in the current cgroup
+func Stop(c *cgroups.Cgroup) error {
+	d, err := getCgroupData(c, 0)
+	if err != nil {
+		return err
+	}
+
+	// first we send SIGTERM to all tasks inside the cgroup
+	if err := Kill(c, syscall.SIGTERM); err != nil {
+		return err
+	}
+
+	// if failed to stop with only SIGTERM, we send SIGKILL
+	if err := waitStop(d, 100*time.Millisecond, 5*time.Second); err != nil {
+		if err := Kill(c, syscall.SIGKILL); err != nil {
+			return err
+		}
+		return waitStop(d, 100*time.Millisecond, -1*time.Second)
```

Comment:
Ideally, we would just try once, wait for some time and then return error if cleanup fails and let the user of libcontainer handle retries and timeouts. We currently do not have a context for exec instances which makes user retries hard. I hope the new API will provide such a context. 

For now, retrying with a backoff sounds like a good strategy. 5s sounds reasonable. We could extend it a minute too. We will have to log and/or surface enough information to let the user take appropriate actions.